jubatus/core/storage/labels.cpp

```diff
@@ -0,0 +1,133 @@
+// Jubatus: Online machine learning framework for distributed environment
+// Copyright (C) 2016 Preferred Networks and Nippon Telegraph and Telephone Corporation.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License version 2.1 as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+#include "labels.hpp"
+
+#include <algorithm>
+#include <string>
+
+#include "jubatus/util/concurrent/lock.h"
+
+namespace jubatus {
+namespace core {
+namespace storage {
+
+labels::labels() {
+}
+
+labels::~labels() {
+}
+
+bool labels::add(const std::string& label) {
+  {
+    util::concurrent::scoped_rlock lock(mutex_);
+    if (master_.find(label) != master_.end()) {
+      return false;
+    }
+
+    if (diff_.find(label) != diff_.end()) {
+      return false;
+    }
+  }
+
+  util::concurrent::scoped_wlock lock(mutex_);
+  diff_[label] = 0;
+  return true;
+}
+
+labels::data_t labels::get() const {
```

Comment:
This method returns the pairs of label and counter.
I rename this to ``get_labels`` .